### PR TITLE
Cleans up redux connect statements

### PR DESCRIPTION
**Pull request by lukaszsobek**
Clean up and order connect statements, remove nesting so redux connect calls are more readable and less error prone. Remove bindActionCreators.

### CODE REVIEW

File: src/components/Editor/Footer.js [hunk 1]
```diff
@@ -199,17 +198,17 @@ class SourceFooter extends PureComponent<Props> {
...
+
+  return {
+    selectedSource,
+    mappedSource: getGeneratedSource(state, source),
+    prettySource: getPrettySource(state, selectedId),
+    endPanelCollapsed: getPaneCollapse(state, "end")
+  };
+};
+
+export default connect(mapStateToProps, actions)(SourceFooter);
```
Comment (wldcordeiro) [inline src/components/Editor/Footer.js]: You seem to jump around between passing in `actions` and a `mapDispatchToProps` object. I kind of prefer the latter as it then limits the number of action creators passed to the component (making it easier to read prop lists in react devtools 😄 )

File: src/components/SecondaryPanes/EventListeners.js [hunk 2]
```diff
@@ -99,20 +98,19 @@ class EventListeners extends Component<Props> {
...
+      ...l,
+      breakpoint: getBreakpoint(state, {
+        sourceId: l.sourceId,
+        line: l.line,
+        column: null
+      })
+    };
+  });
+
+  return { listeners };
```
Comment (wldcordeiro) [inline src/components/SecondaryPanes/EventListeners.js]:
```js
const mapStateToProps = state => ({
  listeners: getEventListeners(state).map(listener => ({
    ...listener,
    breakpoint: getBreakpoint(state, {
      sourceId: listener.sourceId,
      line: listener.line,
      column: null
    })
  }),
});
```

File: src/components/Editor/Footer.js (hunk 1 above)
Comment (lukaszsobek) [reply src/components/Editor/Footer.js]: I agree with the notion, the actions should either be extracted or passed as a whole - having it done a different way every time creates confusion. That is the scope for a different pr however.

The aim here was to clean up the connect, get rid of bindActionCreators etc.

**Comment by wldcordeiro** [comment]
Merge conflicts here. 😦

File: src/components/SecondaryPanes/index.js [hunk 3]
```diff
@@ -400,18 +400,16 @@ SecondaryPanes.contextTypes = {
...
-    breakpointsLoading: getBreakpointsLoading(state),
-    isWaitingOnBreak: getIsWaitingOnBreak(state),
-    shouldPauseOnExceptions: getShouldPauseOnExceptions(state),
-    shouldIgnoreCaughtExceptions: getShouldIgnoreCaughtExceptions(state),
-    workers: getWorkers(state)
-  }),
-  dispatch => bindActionCreators(actions, dispatch)
-)(SecondaryPanes);
+const mapStateToProps = state => ({
+  extra: getExtra(state),
```
Comment (lukaszsobek) [inline src/components/SecondaryPanes/index.js]: Misses expressions

File: src/components/SecondaryPanes/index.js [hunk 4]
```diff
@@ -16,7 +15,6 @@ import {
   getBreakpoints,
   getBreakpointsDisabled,
   getBreakpointsLoading,
-  getExpressions,
   getIsWaitingOnBreak,
```
Comment (lukaszsobek) [inline src/components/SecondaryPanes/index.js]: Misses getExpressions

File: src/components/SecondaryPanes/EventListeners.js (hunk 2 above)
Comment (lukaszsobek) [reply src/components/SecondaryPanes/EventListeners.js]: Done

File: src/components/SecondaryPanes/Breakpoints.js [hunk 5]
```diff
@@ -264,10 +263,9 @@ const _getBreakpoints = createSelector(
...
 );
 
-export default connect(
-  (state, props) => ({
-    breakpoints: _getBreakpoints(state),
-    selectedSource: getSelectedSource(state)
-  }),
-  dispatch => bindActionCreators(actions, dispatch)
-)(Breakpoints);
+const mapStateToProps = (state, props) => ({
```
Comment (lukaszsobek) [inline src/components/SecondaryPanes/Breakpoints.js]: no props used

**Comment by jasonLaster** [comment]
I love this! Lets review/qa this soon so we can land before the conflicts come

**Pull request merged**